testsuite/openshift/objects/auth_config/sections.py

```diff
@@ -208,7 +230,9 @@ def add_simple(self, auth_json: str, name="simple", key="data", **common_feature
     @modify
     def add_json(self, name: str, properties: list[Property], **common_features):
         """Adds json response to AuthConfig"""
-        asdict_properties = [asdict(p) for p in properties]
+        asdict_properties = {}
+        for prop in properties:
+            asdict_properties.update(asdict(prop))
```

Comment:
I do not follow this change, does this mean properties can override each other?

Reply:
I think it's just a way to populate a dict with other dicts in the loop.

*You probably can use something like this to disable overwriting if you want.
```suggestion
            asdict_properties = {**prop, **asdict_properties}
```

Reply:
I meant that before it was a list, meaning all properties were separated, but now it is a one-level dict so that they can override each other. That change is fundamental on the AuthPolicy/AuthConfig level.

Previously we used List because in the CR it was a list, now it is changed to a dict, but we kept the list, which can cause confusion. IF I specify two properties with the same name, I would expect both of them to be propagated to the CR, but now they wont, only one will be.

Basically, I think maybe we should change to dict as well to keep it consistent.

Reply:
Makes sense about possible confusion. Although, the behavior should be the same. @azgabur what would happen if you specify 2 properties with the same name? Would it be overridden / raise an error during creation?

Reply:
Also sidenote: this does not have to solved in this PR specifically, follow-up PR is sufficient

Reply:
I think two properties with same name are no longer possible with the new version - will raise OpenShiftError. Same with other places that it was changed from list to dict.  
Changing the function signature to dict would make object `Property` redundant as the name will now be a key of the dict. I can do the change.